src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp

```diff
@@ -147,80 +141,16 @@ class TypeLookupManager
     void remove_remote_endpoints(
             fastrtps::rtps::ParticipantProxyData* pdata);
 
-    /**
-     * Get the builtin protocols
-     * @return Builtin protocols
-     */
-    fastrtps::rtps::BuiltinProtocols* get_builtin_protocols()
-    {
-        return builtin_protocols_;
-    }
-
-    /**
-     * Get the builtin request writer
-     * @return stateful writer
-     */
-    fastrtps::rtps::StatefulWriter* get_builtin_request_writer();
-
-    /**
-     * Get the builtin reply writer
-     * @return stateful writer
-     */
-    fastrtps::rtps::StatefulWriter* get_builtin_reply_writer();
-
-    /**
-     * Get the builtin request writer's history
-     * @return writer history
-     */
-    fastrtps::rtps::WriterHistory* get_builtin_request_writer_history();
-
-    /**
-     * Get the builtin reply writer's history
-     * @return writer history
-     */
-    fastrtps::rtps::WriterHistory* get_builtin_reply_writer_history();
-
-    /**
-     * Get the builtin request reader
-     * @return stateful reader
-     */
-    fastrtps::rtps::StatefulReader* get_builtin_request_reader();
-
-    /**
-     * Get the builtin reply reader
-     * @return stateful reader
-     */
-    fastrtps::rtps::StatefulReader* get_builtin_reply_reader();
-
-    /**
-     * Get the builtin request reader's history
-     * @return reader history
-     */
-    fastrtps::rtps::ReaderHistory* get_builtin_request_reader_history();
-
-    /**
-     * Get the builtin reply reader's history
-     * @return reader history
-     */
-    fastrtps::rtps::ReaderHistory* get_builtin_reply_reader_history();
-
-    /* TODO Uncomment if security is implemented.
-     #if HAVE_SECURITY
-        bool pairing_remote_reader_with_local_writer_after_security(
-                const fastrtps::rtps::GUID_t& local_writer,
-                const fastrtps::rtps::ReaderProxyData& remote_reader_data);
-
-        bool pairing_remote_writer_with_local_reader_after_security(
-                const fastrtps::rtps::GUID_t& local_reader,
-                const fastrtps::rtps::WriterProxyData& remote_writer_data);
-     #endif
-     */
-
     fastrtps::rtps::SampleIdentity get_type_dependencies(
-            const fastrtps::types::TypeIdentifierSeq& in) const;
+            const fastrtps::types::TypeIdentifierSeq& id_seq) const;
 
     fastrtps::rtps::SampleIdentity get_types(
-            const fastrtps::types::TypeIdentifierSeq& in) const;
+            const fastrtps::types::TypeIdentifierSeq& id_seq) const;
+
+    ReturnCode_t async_get_type(
+            xtypes::TypeInformation typeinformation,
+            GuidPrefix_t type_server,
+            AsyncGetTypeCallback& callback);
```

Comment:
Specifically, this new method should be extensively documented.

Reply:
Added some in [7dafd25](https://github.com/eProsima/Fast-DDS/pull/4126/commits/7dafd25b2d97a27f2f53acf522b0d2c68e72a46e)